### PR TITLE
Internationalize and use more of cjwmodule

### PR DESCRIPTION
**Pull request by giorgio93p**
This pull request accomplishes two related tasks:

- internationalizes the warnings produced by this module, using the utils provided by `cjwmodule`
- uses a `cjwmodule` util that creates some (internationalized) warnings regarding column names

Notice: the corresponding changes to `cjwmodule` are still unpublished: they are pull requests https://github.com/CJWorkbench/cjwmodule/pull/4 and https://github.com/CJWorkbench/cjwmodule/pull/3.

### CODE REVIEW

**Comment by giorgio93p** [comment]
Depends on https://github.com/CJWorkbench/cjwmodule/pull/8

File: test_transpose.py [hunk 1]
```diff
@@ -168,25 +182,43 @@ def test_warn_on_convert_to_str_including_column_header(self):
...
-            result["quick_fixes"],
+            result[1],
             [
                 {
-                    "text": 'Convert "A", "C" to text',
-                    "action": "prependModule",
-                    "args": ["converttotext", {"colnames": ["A", "C"]}],
-                }
+                    "message": i18n_message(
+                        "warnings.headersConvertedToText.error", {"column_name": "A"}
```
Comment (adamhooper) [inline test_transpose.py]: `.message`, not `.error`

File: test_transpose.py [hunk 2]
```diff
@@ -168,25 +182,43 @@ def test_warn_on_convert_to_str_including_column_header(self):
...
+                    ),
+                    "quickFixes": [
+                        {
+                            "text": i18n_message(
+                                "warnings.headersConvertedToText.quick_fix.text",
```
Comment (adamhooper) [inline test_transpose.py]: (while we're editing stuff): `.quickFix`, not `.quick_fix`. (I don't have much of a preference of snake-case vs camel-case, but we shouldn't put both in the very same ID.)

File: transpose.py [hunk 3]
```diff
@@ -136,15 +110,27 @@ def render(table, params, *, input_columns):
...
-        else:
-            colnames = ", ".join(f'"{c}"' for c in to_convert)
-            start = f"Columns {colnames} were"
+        cols_str = ", ".join(f'"{c}"' for c in to_convert)
         warnings.append(
-            f"{start} auto-converted to Text because all columns must have "
-            "the same type."
+            {
+                "message": i18n.trans(
+                    "warnings.differentColumnTypes.error",
```
Comment (adamhooper) [inline transpose.py]: `.message`

File: transpose.py [hunk 4]
```diff
@@ -136,15 +110,27 @@ def render(table, params, *, input_columns):
...
+                    "{n_columns, plural, other {Columns {column_names} were} one {Column {column_names} was}} "
+                    "auto-converted to Text because all columns must have the same type.",
+                    {"n_columns": len(to_convert), "column_names": cols_str},
+                ),
+                "quickFixes": [
+                    {
+                        "text": i18n.trans(
+                            "warnings.differentColumnTypes.quick_fix.text",
```
Comment (adamhooper) [inline transpose.py]: `.quickFix`

File: transpose.py [hunk 5]
```diff
@@ -136,15 +110,27 @@ def render(table, params, *, input_columns):
         # Convert everything to text before converting. (All values must have
         # the same type.)
         to_convert = [c for c in table.columns if input_columns[c].type != "text"]
-        colnames_auto_converted_to_text.extend(to_convert)
-        if len(to_convert) == 1:
-            start = f'Column "{to_convert[0]}" was'
-        else:
-            colnames = ", ".join(f'"{c}"' for c in to_convert)
-            start = f"Columns {colnames} were"
+        cols_str = ", ".join(f'"{c}"' for c in to_convert)
```
Comment (adamhooper) [inline transpose.py]: Let's not do this message formatting. It's not multilingual (and it doesn't even read properly in English.) Instead, please pass arguments `{n_columns}`, `{first_colname}`.

**Review by adamhooper — CHANGES_REQUESTED** [review]
Looking great. Just a few tweaks.

**Review by adamhooper — APPROVED** [review]

**Pull request merged**